### PR TITLE
adds endpoint-specific transforms

### PR DESCRIPTION
**Pull request by thorhj**
Addressing #749 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by thorhj** [comment]
Two considerations for the reviewer:

1. The property for the transform function is called `swaggerTransform` but supports both OpenAPI and Swagger specs of course. Don't know if that is confusion and a better name should be added (or two separate properties).
2. The implementation allows the `swaggerTransform` to be given the value `null` which has its own specific meaning. Maybe this should be the string literal `"disable"` or something else to be more clear.

**Review by Uzlopak — CHANGES_REQUESTED** [review]
missing typing tests.

**Comment by thorhj** [comment]
> missing typing tests.

Added some tests in `types.test.ts`. Let me know if they should be written differently, haven't made such tests before (I'm usually working in Typescript directly 😉)

**Review by mcollina — APPROVED** [review]
lgtm

File: index.d.ts [hunk 1]
```diff
@@ -52,8 +52,26 @@ declare module 'fastify' {
       [statusCode: string]: OpenAPIV3.ResponseObject['links'];
     }
   }
+  
+  interface FastifyContextConfig {
+    swaggerTransform?: SwaggerTransform | null;
+  }
 }
 
+type SwaggerTransform = <S extends FastifySchema = FastifySchema>({
```
Comment (Uzlopak) [inline index.d.ts:61]: Why not
```typescript
type SwaggerTransform<S extends FastifySchema = FastifySchema> = ({
```

And is it still possible to specify the generic?

**Review by Uzlopak — CHANGES_REQUESTED** [review]
I reviewed it now. 

swggerTransform null is quiet unintuitive. I would personally prefer false for deactivating.

File: index.d.ts (hunk 1 above)
Comment (thorhj) [reply index.d.ts:61]: I simply extracted the type from the global transform so I could re-use the same function type. I think it is correct; the function should be generic, not the type.

**Comment by thorhj** [comment]
> I reviewed it now.
> 
> swggerTransform null is quiet unintuitive. I would personally prefer false for deactivating.

I agree, changed it to `false` instead of `null`.

File: index.d.ts (hunk 1 above)
Comment (Uzlopak) [reply index.d.ts:61]: Thanks for the clarification. Lets see if somebody complains... :))

**Comment by fredrikj31** [comment]
When is this set to be released? 🤔

**Comment by mcollina** [comment]
done now

**Pull request merged**